pkg/cmd/describe/describe_rollout.go

```diff
@@ -0,0 +1,329 @@
+
+	rollout, err := o.RolloutViewerFn(info.Object)
+	if err != nil {
+		return err
+	}
+
+	o.printRolloutInfo(rollout)
+	return nil
+}
+
+func (o *DescribeRolloutOptions) watchRollout(r *resource.Result) error {
+	infos, err := r.Infos()
+	if err != nil {
+		return err
+	}
+	if len(infos) != 1 {
+		return fmt.Errorf("watch is only supported on a single rollout")
+	}
+	info := infos[0]
+
+	watcher, err := o.RolloutsClient.Watch(context.TODO(), metav1.ListOptions{
+		FieldSelector: "metadata.name=" + info.Name,
+	})
+
+	if err != nil {
+		return err
+	}
+	defer watcher.Stop()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	if o.TimeoutSeconds > 0 {
+		ctx, cancel = context.WithTimeout(ctx, time.Duration(o.TimeoutSeconds)*time.Second)
+		defer cancel()
+	}
+
+	return o.watchRolloutUpdates(ctx, watcher)
+}
+
+func (o *DescribeRolloutOptions) watchRolloutUpdates(ctx context.Context, watcher watch.Interface) error {
+	for {
+		select {
+		case event, ok := <-watcher.ResultChan():
+			if !ok {
+				return nil
+			}
+			if event.Type == watch.Added || event.Type == watch.Modified {
+				if rollout, ok := event.Object.(*rolloutsapi.Rollout); ok {
+					o.clearScreen()
+					o.printRolloutInfo(rollout)
+				}
+			}
+		case <-ctx.Done():
+			return ctx.Err()
+		}
+	}
+}
+
+func (o *DescribeRolloutOptions) clearScreen() {
+	fmt.Fprint(o.Out, "\033[2J\033[H")
+}
+
+func (o *DescribeRolloutOptions) GetResources(rollout *rolloutsapi.RolloutSpec) (*WorkloadInfo, error) {
+	resources := []string{rollout.WorkloadRef.Kind + "/" + rollout.WorkloadRef.Name}
+	r := o.Builder().
+		WithScheme(internalapi.GetScheme(), scheme.Scheme.PrioritizedVersionsAllGroups()...).
+		NamespaceParam(o.Namespace).DefaultNamespace().
+		ResourceTypeOrNameArgs(true, resources...).
+		ContinueOnError().
+		Latest().
+		Flatten().
+		Do()
+
+	if err := r.Err(); err != nil {
+		return nil, err
+	}
+
+	obj, err := r.Object()
+	if err != nil {
+		return nil, err
+	}
+
+	info := &WorkloadInfo{}
+	objValue := reflect.ValueOf(obj).Elem()
+	info.Name = objValue.FieldByName("Name").String()
+	info.Kind = objValue.Type().Name()
+
+	podTemplateSpec := objValue.FieldByName("Spec").FieldByName("Template").FieldByName("Spec")
+	containers := podTemplateSpec.FieldByName("Containers")
+	for i := 0; i < containers.Len(); i++ {
+		container := containers.Index(i)
+		info.Images = append(info.Images, container.FieldByName("Image").String())
+	}
+
+	spec := objValue.FieldByName("Spec")
+	status := objValue.FieldByName("Status")
+
+	if replicas := spec.FieldByName("Replicas"); replicas.IsValid() && !replicas.IsNil() {
+		info.Replicas.Desired = int32(replicas.Elem().Int())
+	}
+
+	info.Replicas.Current = int32(status.FieldByName("Replicas").Int())
+	info.Replicas.Updated = int32(status.FieldByName("UpdatedReplicas").Int())
+	info.Replicas.Ready = int32(status.FieldByName("ReadyReplicas").Int())
+	info.Replicas.Available = int32(status.FieldByName("AvailableReplicas").Int())
```

Comment:
plz check the rollout.status.ObservedGeneration against rollout.Metadata.generation, if two generation does not match, we shall not display status related fields.